### DISCUSSION TITLE
Is there a way to limit the chunk size returned from BodyStream?

### DISCUSSION THREAD

**Discussion by oliverb123** [tokio-rs/axum#1682] [Q&A] [answered]
Hi! Really enjoying using axum, but I've hit a bit of a stumbling block here. 

I'm implementing code like this:
```
#[async_trait::async_trait]
impl Storeable for BodyStream {
    async fn store(&mut self, mut handle: WriteHandle) -> Result<ContentStored, StorageError> {
        while let Some(chunk) = self.next().await.transpose()? {
            handle.write_all(&chunk).await?;
        }
        let info = handle.info();
        let id = handle.close().await?;
        Ok(ContentStored {
            content_id: id,
            storage_driver: info,
        })
    }
}
```

And really wishing I could have some strong guarantee about the size of `chunk` - maybe I'm missing something obvious, but I'm (basically) looking for strong control over the maximum amount of memory any given request can take up while I'm streaming it's body somewhere.

A cursory google failed to turn up any likely solutions for me. I'm guessing I could implement this with some middleware, but I'd prefer not to if there's a simpler way (I'm honestly not super familar with the tower).

**Comment by davidpdrsn** [comment, marked as answer]
This is more a job for hyper than axum. Perhaps https://docs.rs/hyper/0.14.23/hyper/server/struct.Builder.html#method.http1_max_buf_size is what you want.

**Reply by bonsairobo** [reply]
How would one apply this in an axum server?

**Comment by oliverb123** [comment]
Perfect, thank you!
